static/src/js/components/KeywordForm/__tests__/KeywordForm.test.jsx

```diff
@@ -109,3 +158,177 @@ describe('when initialData prop changes', () => {
     expect(screen.getByDisplayValue('Updated Keyword')).toBeInTheDocument()
   })
 })
+
+// New
+describe('when the form is submitted', () => {
+  test('should open the modal when Save button is clicked', async () => {
+    const user = userEvent.setup()
+    render(<KeywordForm
+      initialData={mockInitialData}
+      scheme={{ name: 'sciencekeywords' }}
+      version={{ version: 'draft' }}
+    />)
+
+    // Find the Save button within the form
+    const saveButton = screen.getByRole('button', { name: 'Save' })
+    await user.click(saveButton)
+
+    // Check for the presence of the modal
+    expect(screen.getByRole('textbox', { name: /Add a user note for your change/i })).toBeVisible()
+  })
+
+  test('should call createUpdateKmsConcept when final save is clicked', async () => {
+    const user = userEvent.setup()
+    const scheme = { name: 'sciencekeywords' }
+    const version = { version: 'draft' }
+    const mockToken = 'mock-token'
+    const mockUid = 'test-user-id'
+    const mockOnSave = vi.fn()
+
+    render(<KeywordForm
+      initialData={mockInitialData}
+      scheme={scheme}
+      version={version}
+      onSave={mockOnSave}
+      token={mockToken}
+      uid={mockUid}
+    />)
+
+    // Click the form's Save button
+    await user.click(screen.getByRole('button', { name: 'Save' }))
+
+    // Click the modal's Save button
+    const modalSaveButton = screen.getAllByRole('button', { name: 'Save' })[1]
+    await user.click(modalSaveButton)
+
+    await waitFor(() => {
+      expect(convertFormDataToRdf).toHaveBeenCalledWith(mockInitialData, '', scheme, mockUid)
+    })
+
+    const expectedRdfXml = {
+      ...mockInitialData,
+      userNote: '',
+      scheme,
+      rdf: true
+    }
+
+    expect(createUpdateKmsConcept).toHaveBeenCalledWith({
+      rdfXml: expectedRdfXml,
+      version,
+      token: mockToken
+    })
+  })
+
+  test('should close the modal after successful save', async () => {
+    const user = userEvent.setup()
+    render(<KeywordForm
```

Comment:
We could just make a `const setup = (overrideProps = {}) => {` method and pull these renders behind that

Reply:
Done